2] loto/lotto.py

```diff
@@ -0,0 +1,82 @@
+"""
+Workshop_01 - Program_02
+---------------
+LOTTO Simulator
+---------------
+(c) Tomas Dolejsek 2024-01-30
+
+LOTTO is a numeric game that involves drawing 6 numbers from the range of 1 – 49. The player's task is to correctly
+guess the drawn numbers. You are rewarded if you correctly match 3, 4, 5 or 6 numbers.
+
+Write a program that:
+
+1. asks user to select 6 numbers, while checking the following conditions:
+    - whether the string entered is a valid number,
+    - whether the user has not entered a given number before,
+    - if the number is in the range of 1-49,
+2. after entering 6 numbers, sorts them in ascending order and displays them on the screen,
+3. draws 6 numbers from the range and displays them on the screen,
+4. informs the player how many numbers they have matched.
+"""
+
+from random import shuffle
+
+
+class InvalidNumberError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+def get_numbers() -> list[int]:
+    """
+    Ask user for six numbers. For each number validate its type (int), range (1-49)
+    and whether it hasn't been taken before.
+    :return: list of user's numbers
+    """
+    n = 0
+    numbers = []
+    while n < 6:
+        try:
+            number = int(input(f"Enter {n + 1}. number (1 - 49): "))
+            if not (1 <= number <= 49):
+                raise InvalidNumberError("The number is out of range! Try again.")
+            if number in numbers:
+                raise InvalidNumberError("You've already picked that number! Try again.")
```

Comment:
Neni dobry mit pro dva ruzny pripady stejnou eror clasu = kdyz uz si das tu praci to takle delat, tak udelaj dve ruzne. jedna pro kupr OutOfRangeErr  druha kupr pro DupliceErr.